### PR TITLE
Updated API Getting Started

### PR DESCRIPTION
**Pull request by Sinacam**
Includes a new linear tutorial and moving neural network tutorial to a standalone page.
Some links are "placeholder" currently because the pages don't exist yet.

### CODE REVIEW

File: docs/api/linear_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+The preprocessor handles many issues such as: mapping
+the labels into indices and transforming textual data to
+numerical data.
+
+Next we train the model:
+
+.. literalinclude:: ../examples/linear_quickstart.py
+    :language: python3
+    :lines: 8-10
+
```
Comment (cjlin1) [inline docs/api/linear_tutorial.rst:53]: Need to explain  y and x

File: docs/api/linear_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+Where the third argument is the options string for
```
Comment (cjlin1) [inline docs/api/linear_tutorial.rst:54]: We don't have ", where" here. The "Where" isn't good. Can just say "The third argument ..."

File: docs/api/linear_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+`LibLinear <https://www.csie.ntu.edu.tw/~cjlin/liblinear/>`_.
+We may leave it as the default for now.
+
+Once we have the model, we may predict with it:
+
+.. literalinclude:: ../examples/linear_quickstart.py
+    :language: python3
+    :lines: 12
+
+Where :py:`preds` holds the decision values, i.e. the raw values
```
Comment (cjlin1) [inline docs/api/linear_tutorial.rst:64]: same issue

File: docs/api/linear_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+outputted by the model. To transform it into predictions,
+the simplest way is to take the positive values as positive
+predictions, i.e. :py:`preds > 0`.
```
Comment (cjlin1) [inline docs/api/linear_tutorial.rst:67]: this sentence doesn't read well. "positive predictions" not clear. rewrite it a bit

**Comment by Sinacam** [comment]
> 

My mistake, it is an unrelated planned modification.

**Review by cjlin1 — DISMISSED** [review]

**Comment by cjlin1** [comment]
Merged but got conflicts. Please fix it

**Comment by Sinacam** [comment]
The conflict was a 2-line offset for quickstart.py, which required modifications in the docs referencing it.
I rebased onto the new master because it would be faulty to only fix the line numbers but not include the changes to quickstart.py.

**Review by cjlin1 — APPROVED** [review]

**Pull request merged**